Sources/Resolver/Resolver.swift

```diff
@@ -319,6 +355,31 @@ public final class Resolver {
         return nil
     }
 
+    /// Internal function searches the current and child registries for all ResolverRegistration<Service>s that matches
+    /// the supplied type.
+    private final func lookupAll<Service>(_ type: Service.Type) -> [ResolverRegistration<Service>]? {
+        guard let values = lookupAllKeyed(type)?.values else {
+            return nil
+        }
+        return Array(values)
+    }
+
+    /// Internal function searches the current and child registries for all ResolverRegistration<Service>s that matches
+    /// the supplied type combind with their registered names.
+    private final func lookupAllKeyed<Service>(_ type: Service.Type) -> [String: ResolverRegistration<Service>]? {
+        let key = Int(bitPattern: ObjectIdentifier(Service.self))
```

Comment:
Note this should be `ObjectIdentifier(type)` as we should have been using the passed type instead of attempting to infer the type once more.

Reply:
Fixed!